superset-frontend/src/views/CRUD/annotation/AnnotationModal.tsx

```diff
@@ -206,32 +225,38 @@ const AnnotationModal: FunctionComponent<AnnotationModalProps> = ({
   };
 
   // Initialize
-  if (
-    isEditMode &&
-    (!currentAnnotation ||
-      !currentAnnotation.id ||
-      (annotation && annotation.id !== currentAnnotation.id) ||
-      (isHidden && show))
-  ) {
-    if (annotation && annotation.id !== null && !loading) {
-      const id = annotation.id || 0;
-
-      fetchResource(id).then(() => {
-        setCurrentAnnotation(resource);
+  useEffect(() => {
+    if (
+      isEditMode &&
+      (!currentAnnotation ||
+        !currentAnnotation.id ||
+        (annotation && annotation.id !== currentAnnotation.id) ||
+        (isHidden && show))
+    ) {
```

Comment:
Maybe we can split these `if` conditions into multiple early returns. Currently it's kind of hard to reason with.

Reply:
you can also use optional chaining to help if you want.